spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/views/CreateIcebergView.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.catalyst.plans.logical.views
+
+import org.apache.spark.sql.catalyst.plans.logical.BinaryCommand
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+
+case class CreateIcebergView(
```

Comment:
I think this should have:
* `child: LogicalPlan`
* `queryText: String` (should not be optional)
* `query: LogicalPlan`
* `columnAliases: Seq[String]` (translate from Spark's weirdness in conversion)
* `columnComments: Seq[Option[String]]`
* `comment: String`
* `properties: Map[String, String]`
* `allowExisting: Boolean`
* `replace: Boolean`

Reply:
this makes sense, I've updated this accordingly, thanks.

Reply:
@rdblue the only modification I had to do here is to keep `queryColumnNames` since we need to pass that through to the catalog. `queryColumnNames` are known once the `query` is resolved in `RewriteViews`, which is also the place where these are set from `query.schema.fieldNames`